exetera/core/fields.py

```diff
@@ -243,7 +264,11 @@ def dtype(self):
         return self._dataset.dtype
 
     def __getitem__(self, item):
-        return self._dataset[item]
+        if self._field_instance.filter is not None:
+            mask = self._field_instance.filter[:]
```

Comment:
Apply the item to select indices from the mask first then use these to get values from self._dataset:
```
idx = mask[item]
```